Overview page for Run your dbt projects (#2168)
fishtown-analytics/docs.getdbt.com: website/docs/docs/develop/run-your-dbt-projects.md

---
title: "Run your dbt projects"
id: "run-your-dbt-projects"
---
You can run your dbt projects with [dbt Cloud](/docs/develop/dbt-cloud-features) and [dbt Core](https://github.com/dbt-labs/dbt-core). dbt Cloud is a hosted application where you can develop directly from a web browser. dbt Core is an open source project where you can develop from the command line.

> Among other features, dbt Cloud provides a hosted development environment to help you build, test, run, and version control your project faster. It also includes a purpose-built scheduler and an easier way to share your [dbt project's documentation](/docs/collaborate/build-and-view-your-docs) with your team. These development tasks are directly built into dbt Cloud. See [dbt Cloud IDE] (link to one of the IDE pages mirna is working on) for more details.

Author: hey @mirnawong1 : i think more links to IDE docs could be added in this paragraph. let's chat more! 🙏

> Among other features, dbt Cloud provides a hosted development environment to help you build, test, run, and version control your project faster. It also includes a purpose-built scheduler and an easier way to share your [dbt project's documentation](/docs/collaborate/build-and-view-your-docs) with your team. These development tasks are directly built into dbt Cloud. See [dbt Cloud IDE] (link to one of the IDE pages mirna is working on) for more details.

Reviewer: updated links and sugg IDE instead of hosted

```suggestion
Among other features, dbt Cloud provides an integrated development environment (IDE) to help you build, test, run, and version control your project faster. It also includes a purpose-built scheduler and an easier way to share your [dbt project's documentation](/docs/collaborate/build-and-view-your-docs) with your team. These development tasks are directly built into dbt Cloud. Review [dbt Cloud IDE] (/docs/develop/develop-in-the-cloud) for more details.
```

Reviewer: (/docs/develop/develop-in-the-cloud) changing to this link now

With dbt Core, you can run your dbt projects from the command line. The command line interface (CLI) is available from your computer's terminal application such as Terminal and iTerm. When using the command line, you can run commands and do other work from the current working directory on your computer. Before running the dbt project from the command line, make sure you are working in your dbt project directory. Learning terminal commands such as `cd` (change directory), `ls` (list directory contents), and `pwd` (present working directory) can help you navigate the directory structure on your system.

> When working with a project, the dbt commands you commonly use are:

Reviewer: i'm suggesting being explicit here with the cli and ide

```suggestion
When working with a project in the CLI or Cloud IDE, the dbt commands you commonly use are:
```

Author: i don't think the sentence is confusing or ambiguous without "in the CLI or Cloud IDE" so wanna lean toward being concise (fewer words). 
 
do you find it unclear?

Reviewer: oh yea i did initially - i think it's because it followed the dbt core summary maybe? so i had to re read to make sure the commands were related to both

- [dbt run](/reference/commands/run) &mdash; Runs the models you defined in your project
- [dbt build](/reference/commands/build) &mdash; Builds and tests your selected resources such as models, seeds, snapshots, and tests
- [dbt test](/reference/commands/test) &mdash; Executes the tests you defined for your project

For information on all dbt commands and their arguments (flags), see the [dbt command reference](https://docs.getdbt.com/reference/dbt-commands). If you want to list all dbt commands from the command line, run `dbt --help`. To list a dbt command’s specific arguments, run `dbt COMMAND_NAME --help` .

## Related docs

- [How we set up our computers for working on dbt projects](https://discourse.getdbt.com/t/how-we-set-up-our-computers-for-working-on-dbt-projects/243)

> - [Model selection syntax](https://docs.getdbt.com/reference/node-selection/syntax)

Reviewer: suggesting some more docs on the scheduler (i linked the dbt cloud features page - is this right?), ide, version control. 

```suggestion
- [Model selection syntax](https://docs.getdbt.com/reference/node-selection/syntax)
- [Version control basics](/docs/collaborate/git/version-control-basics)
- [dbt Cloud tips](/docs/develop/dbt-cloud-tips)
```

---
title: "Run your dbt projects"
id: "run-your-dbt-projects"
---
You can run your dbt projects with [dbt Cloud](/docs/develop/dbt-cloud-features) and [dbt Core](https://github.com/dbt-labs/dbt-core). dbt Cloud is a hosted application where you can develop directly from a web browser. dbt Core is an open source project where you can develop from the command line.

> Among other features, dbt Cloud provides a development environment to help you build, test, run, and [version control](/docs/collaborate/git-version-control) your project faster. It also includes an easier way to share your [dbt project's documentation](/docs/collaborate/build-and-view-your-docs) with your team. These development tasks are directly built into dbt Cloud for an _integrated development environment_ (IDE). For more details, see the [Cloud IDE features](/docs/develop/dbt-cloud-features#ide-features).

Author: hi @mirnawong1 , this paragraph is introducing the IDE so would like to link out to more information about all the features in the IDE. i think the "Develop in the Cloud" is helpful to reference in this page too; i've added it to the "Related docs" section as an escape hatch to it

is `/docs/develop/dbt-cloud-features#ide-features` the correct anchor link to use for the **IDE features** section?

Reviewer: /docs/develop/dbt-cloud-features.md#ide-features is correct

With dbt Core, you can run your dbt projects from the command line. The command line interface (CLI) is available from your computer's terminal application such as Terminal and iTerm. When using the command line, you can run commands and do other work from the current working directory on your computer. Before running the dbt project from the command line, make sure you are working in your dbt project directory. Learning terminal commands such as `cd` (change directory), `ls` (list directory contents), and `pwd` (present working directory) can help you navigate the directory structure on your system.

> When you run your dbt project, the commands you commonly use are:

Reviewer 2: This felt a little Core focused, but it's true for either product. Maybe a tiny tweak will clarify
```suggestion
When running your dbt project from dbt Core or dbt Cloud, you'll commonly use:
```

Author: ```suggestion
When running your project from dbt Core or dbt Cloud, the commands you commonly use are:
```

Author: @mirnawong1 also wanted to extend the first clause! 🙇‍♀️

- [dbt run](/reference/commands/run) &mdash; Runs the models you defined in your project
- [dbt build](/reference/commands/build) &mdash; Builds and tests your selected resources such as models, seeds, snapshots, and tests
- [dbt test](/reference/commands/test) &mdash; Executes the tests you defined for your project

For information on all dbt commands and their arguments (flags), see the [dbt command reference](https://docs.getdbt.com/reference/dbt-commands). If you want to list all dbt commands from the command line, run `dbt --help`. To list a dbt command’s specific arguments, run `dbt COMMAND_NAME --help` .

## Related docs

- [How we set up our computers for working on dbt projects](https://discourse.getdbt.com/t/how-we-set-up-our-computers-for-working-on-dbt-projects/243)
- [Model selection syntax](https://docs.getdbt.com/reference/node-selection/syntax)

> - [Develop in the Cloud](/docs/develop/develop-in-the-cloud)

Author: @mirnawong1 , added the "Develop in the Cloud" to related docs
